plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -303,6 +303,9 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
     private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
         IcebergTableName name = IcebergTableName.from(tableName.getTableName());
+        if (name.getTableType() == DATA) {
+            return Optional.empty();
```

Comment:
We can also remove the metastore call below if we translate the not found exception into empty.

Reply:
Right, will handle in a follow up, since this also requires more test coverage (https://github.com/trinodb/trino/issues/8690)

Reply:
Addressed in https://github.com/trinodb/trino/pull/8692